### PR TITLE
fix minor bug introduced by last refactor and add test to cover

### PR DESCRIPTION
**Pull request by turino**
`unconvinced_legislators` was returning ineligible senators

also, there were some tests in location_spec that weren't really testing anything

### CODE REVIEW

File: app/models/zip_code.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@ def single_district?
   end
 
   def single_district
-    single_district? && districts.first
+    districts.first if single_district?
   end
```
Comment (turino) [inline app/models/zip_code.rb]: I guess I went overboard changing `a if a` to `a && a` -- in this case I want it to return `nil`, not `false`

File: spec/models/location_spec.rb [hunk 2]
```diff
@@ -26,114 +26,111 @@
...
-    let(:district) { FactoryGirl.create(:district) }
-    let(:person)   { FactoryGirl.create(:person, district: district,
-                                                 state:    district.state,
-                                                 zip_code: '03431') }
+    let(:location) { FactoryGirl.create(:location, zip_code: '03431') }
     context "bad address" do
       it "doesn't change location" do
-        expect {
-          person.update_location(address: '2020 Oregon St', zip: 'bad')
-        }.to_not change { person.location }
```
Comment (turino) [inline spec/models/location_spec.rb]: this was always returning true, whether location changed or not

File: app/models/person.rb [hunk 3]
```diff
@@ -59,7 +59,7 @@ def next_target
   end
 
   def unconvinced_legislators
-    legislators && legislators.unconvinced
+    legislators && legislators.unconvinced.eligible
   end
```
Comment (Rio517) [inline app/models/person.rb]: I'm wondering if we can save db calls by making these associations - but not overload the class.

Comment (turino) [inline app/models/person.rb]: yeah, sounds like a good thing to investigate. I'm eager to learn as much as I can about performance optimization -- not something I had to worry about on my last couple projects

**Pull request merged**